rootfs/api/models/app.py

```diff
@@ -393,13 +393,14 @@ def _default_structure(self, release):
         if not release.build.sha:
             structure = {'cmd': 1}
 
-        # if a dockerfile exists without a procfile, assume docker workflow
-        elif release.build.dockerfile and not release.build.procfile:
+        # if a dockerfile, assume docker workflow
+        elif release.build.dockerfile:
             structure = {'cmd': 1}
 
-        # if a procfile exists without a web entry, assume docker workflow
+        # if a procfile exists without a web entry,
+        # assume heroku workflow with nothing to scale by default
         elif release.build.procfile and 'web' not in release.build.procfile:
-            structure = {'cmd': 1}
+            structure = {}
```

Comment:
Ah, I see what you are going for. Why wouldn't we grab the type in `Procfile` and scale that to 1? In case we have more than 1 type? I'm going on a limb that not auto picking a type is how Heroku behaves?